torchrl/objectives/ddpg.py

```diff
@@ -293,7 +293,9 @@ def _loss_actor(
         self,
         tensordict: TensorDictBase,
     ) -> torch.Tensor:
-        td_copy = tensordict.select(*self.actor_in_keys).detach()
+        td_copy = tensordict.select(
+            *self.actor_in_keys, *self.value_network.in_keys
```

Comment:
The value net takes as input the output of the actor network, so it could be that the out_keys of the actor are not there yet. Either set `strict=False` (dangerous, because we could end up having a long trace stack in the error when calling actor_network because of a missing key) or do this

```python
        td_copy = tensordict.select(
            *self.actor_in_keys
        ).detach()
        td_copy = self.actor_network(
            td_copy,
            params=self.actor_network_params,
        )
        # check if the critic has some keys that can't be found in in or out_keys from actor
        critic_exclusive = (set(self.actor_in_keys) | set(self.actor_in_keys)) ^ set(self.value_network.in_keys)
        if critic_exclusive:
            td_copy = td_copy.update(tensordict.select(*critic_exclusive))
        td_copy = self.value_network(
            td_copy,
            params=self._cached_detached_value_params,
        )
```

Reply:
the actor is called in the next line of code, so its keys are assured to be there no?
i do not understand the problem and I would like to avoid adding further select and updates
can you explaint a bit more what you mean?